src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -62,9 +66,9 @@ makeRawPage ba off = RawPage (div2 off) ba
 
 rawPageEntry
     :: RawPage
-    -> P.Vector Word8 -- ^ key
-    -> Maybe (Entry (P.Vector Word8) BlobSpan)
-rawPageEntry !page !key = bisect 0 (fromIntegral dirNumKeys)
+    -> SerialisedKey
+    -> Maybe (Entry SerialisedValue BlobSpan)
+rawPageEntry !page (SerialisedKey' !key) = coerce $ bisect 0 (fromIntegral dirNumKeys)
```

Comment:
Can we use proper constructors please, not `coerce`.

Reply:
In general we cannot, this coerces from `Maybe (Entry (P.Vector Word8) BlobSpan` to `Maybe (Entry SerialisedValue BlobSpan)`.

But in this specific case, sure, we won't need to `coerce` if `bisect`, `found` etc returned types with `SerialisedValue`

Reply:
I didn't want to cause too much churn in this PR, but I'm happy to change everything to `Serialised*` constructors